// react-chat-app/lib/cjs/Avatar.js
'use strict';

var React = require('react');
var styledComponents_esm = require('./styled-components.esm-92a8efa7.js');
var theme = require('./theme-37c6f16c.js');
var style = require('./style-4ef8db8a.js');
require('./mixins-4c7023a3.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

function Avatar({
  src,
  size = '48px',
  status,
  StatusIconSize = '8px',
  ...rest
}) {
  return /*#__PURE__*/React__default['default'].createElement(styledComponents_esm.Le, {
    theme: theme.theme
  }, /*#__PURE__*/React__default['default'].createElement(style.AvatarWrapper, rest, status && /*#__PURE__*/React__default['default'].createElement(style.StatusIcon, {
    status: status,
    size: StatusIconSize
  }), /*#__PURE__*/React__default['default'].createElement(style.AvatarClip, {
    size: size
  }, /*#__PURE__*/React__default['default'].createElement(style.AvatarImage, {
    src: src,
    alt: ""
  }))));
}

Avatar.propTypes = {
  src: styledComponents_esm.propTypes.string,
  size: styledComponents_esm.propTypes.string,
  status: styledComponents_esm.propTypes.oneOf(['online', 'offline']),
  StatusIconSize: styledComponents_esm.propTypes.string
};
Avatar.defaultProps = {
  size: '48px',
  status: 'online',
  StatusIconSize: '8px'
};

module.exports = Avatar;
