// react-chat-app/lib/cjs/Badge.js
'use strict';

var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-9bd7278f.js');
var React = require('react');
var styledComponents_esm = require('./styled-components.esm-92a8efa7.js');
var theme = require('./theme-37c6f16c.js');
var mixins = require('./mixins-4c7023a3.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

const varients = {
  dot: styledComponents_esm._e`
  padding: 5px;
  position: relative;
  &::after {
      content: "";
      ${({
    theme
  }) => mixins.circle(theme.red, '6px')}
       display: ${({
    show
  }) => show ? 'block' : 'none'};
       position: absolute; 
       right: 0;
       top: 0;
  }
  `,
  default: styledComponents_esm._e`
    ${({
    theme
  }) => mixins.circle(theme.red, '26px')}
    display: flex;
    align-items: center;
    justify-content: center;
    box-shadow: 0px 18px 40px 0px rgba(0, 0, 0, 0.04),
      0px 6px 12px 0px rgba(0, 0, 0, 0.08);
    ${({
    showZero,
    count
  }) => !showZero && count === 0 && 'visibility: hidden'}
  `
};
const Count = styledComponents_esm.Ye.div`
  font-size: ${({
  theme
}) => theme.normal};
  color: #fff;
`;
const badageWrapper = styledComponents_esm.Ye.div`
  display: inline-block;
  ${({
  varient
}) => varients[varient]}
`;

function Badge({
  children,
  show = false,
  count = 0,
  size = '26px',
  showZero = false,
  ...rest
}) {
  return /*#__PURE__*/React__default['default'].createElement(styledComponents_esm.Le, {
    theme: theme.theme
  }, /*#__PURE__*/React__default['default'].createElement(badageWrapper, _rollupPluginBabelHelpers._extends({
    varient: children ? 'dot' : 'default',
    show: show,
    count: count,
    size: size,
    showZero: showZero
  }, rest), children || /*#__PURE__*/React__default['default'].createElement(Count, null, count)));
}

Badge.propTypes = {
  children: styledComponents_esm.propTypes.any,
  show: styledComponents_esm.propTypes.bool,
  count: styledComponents_esm.propTypes.number,
  showZero: styledComponents_esm.propTypes.bool,
  size: styledComponents_esm.propTypes.string
};

module.exports = Badge;
