// react-chat-app/lib/cjs/Bubble.js
'use strict';

var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-9bd7278f.js');
var React = require('react');
var styledComponents_esm = require('./styled-components.esm-92a8efa7.js');
var theme = require('./theme-37c6f16c.js');
var Paragraph = require('./Paragraph.js');
var Icon = require('./index-4c52fc8e.js');
var Text = require('./Text.js');
require('./style-0ac2425a.js');
require('./style-94d0e43c.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

function _interopNamespace(e) {
  if (e && e.__esModule) return e;
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () {
            return e[k];
          }
        });
      }
    });
  }
  n['default'] = e;
  return Object.freeze(n);
}

var React__namespace = /*#__PURE__*/_interopNamespace(React);
var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

var _path;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function SvgBubbleTip(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends({
    width: 29,
    height: 16,
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path || (_path = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M8.488 1.146C8.165 2.53 6.728 6.538.511 10.919a.989.989 0 00.356 1.774c.034.008.867.19 2.364.32 4.266.374 13.06.29 24.214-4.426.82.116-23.067-10.41-18.957-7.441z",
    fill: "#fff"
  })));
}

const Time = styledComponents_esm.Ye(Paragraph).attrs({
  type: 'secondary',
  size: 'small'
})`
  margin: 6px;
  margin-left: 24px;
  word-spacing: 1rem;
`;
const BubbleTip = styledComponents_esm.Ye(Icon.Icon)`
  position: absolute;
  bottom: -15%;
  left: 0;
  z-index: 5;
`;
const BubbleWrapper = styledComponents_esm.Ye.div`
  padding: 15px 22px;
  box-shadow: 0px 8px 24px rgba(0, 0, 0, 0.1);
  border-radius: 100px;
  position: relative;
  z-index: 10;
`;
const MessageText = styledComponents_esm.Ye(Text)``;
const typeVariants = {
  from: styledComponents_esm._e`
    align-self: flex-end;
    ${BubbleWrapper} {
      background-color: ${({
    theme
  }) => theme.primaryColor};
    }
    ${BubbleTip} {
      transform: rotateY(180deg);
      left: unset;
      right: 0;
      /* svg color */
      path {
        fill: ${({
    theme
  }) => theme.primaryColor};
      }
    }
    ${Time} {
      text-align: right;
      margin-left: 0;
      margin-right: 24px;
    }
    ${MessageText} {
      color: white;
    }
  `
};
const StyledBubble = styledComponents_esm.Ye.div`
  display: flex;
  flex-direction: column;
  ${({
  type
}) => type && typeVariants[type]}
`;

function Bubble({
  children,
  type,
  time,
  ...rest
}) {
  return /*#__PURE__*/React__default['default'].createElement(styledComponents_esm.Le, {
    theme: theme.theme
  }, /*#__PURE__*/React__default['default'].createElement(StyledBubble, _rollupPluginBabelHelpers._extends({
    type: type
  }, rest), /*#__PURE__*/React__default['default'].createElement(BubbleWrapper, null, /*#__PURE__*/React__default['default'].createElement(BubbleTip, {
    icon: SvgBubbleTip,
    width: 40,
    height: 28,
    color: "white"
  }), /*#__PURE__*/React__default['default'].createElement(MessageText, null, children)), /*#__PURE__*/React__default['default'].createElement(Time, null, time)));
}

Bubble.propTypes = {
  children: styledComponents_esm.propTypes.any,
  type: styledComponents_esm.propTypes.oneOf(['from', 'to']),
  time: styledComponents_esm.propTypes.string
};

module.exports = Bubble;
