// react-chat-app/lib/cjs/ChatApp.js
'use strict';

var React = require('react');
var styledComponents_esm = require('./styled-components.esm-92a8efa7.js');
var index = require('./index-e429b214.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

const Nav = styledComponents_esm.Ye.nav`
  flex-shrink: 0;

  position: relative;
  z-index: 100;
`;
const Sidebar = styledComponents_esm.Ye.aside`
  max-width: 448px;
  min-width: 344px;
  height: 100vh;
  flex: 1;
  background: ${({
  theme
}) => theme.grediantGray};

  position: relative;
  z-index: 50;
  > div {
    will-change: transform, opacity;
    position: absolute;
    width: 100%;
  }
`;
const Content = styledComponents_esm.Ye.main`
  flex: 2;
  position: relative;
`;
const StyledChatApp = styledComponents_esm.Ye.div`
  display: flex;
  height: 100vh;
  width: 100vw;
  overflow: hidden;
  position: relative;
`;

function ChatApp({
  children,
  navBar: NavBar,
  sideBar: MessageList,
  content: Conversation,
  profile: Profile,
  showProfile = true,
  ...rest
}) {
  const animeProps = showProfile ? index.useSpring({
    opacity: 1,
    transform: 'translate3d(0px, 0px, 0px)',
    from: {
      opacity: 0,
      transform: 'translate3d(-100px, 0px, 0px)'
    },
    delay: 500
  }) : {
    width: 0
  };
  return /*#__PURE__*/React__default['default'].createElement(StyledChatApp, rest, /*#__PURE__*/React__default['default'].createElement(Nav, null, NavBar && /*#__PURE__*/React__default['default'].createElement(NavBar, null)), /*#__PURE__*/React__default['default'].createElement(Sidebar, null, MessageList && /*#__PURE__*/React__default['default'].createElement(MessageList, null)), /*#__PURE__*/React__default['default'].createElement(Content, null, Conversation && /*#__PURE__*/React__default['default'].createElement(Conversation, null)), /*#__PURE__*/React__default['default'].createElement(index.animated.div, {
    style: { ...animeProps
    }
  }, Profile && /*#__PURE__*/React__default['default'].createElement(Profile, null)));
}

ChatApp.propTypes = {
  children: styledComponents_esm.propTypes.any,
  navBar: styledComponents_esm.propTypes.any,
  sideBar: styledComponents_esm.propTypes.any,
  content: styledComponents_esm.propTypes.any,
  profile: styledComponents_esm.propTypes.any,
  showProfile: styledComponents_esm.propTypes.bool
};

module.exports = ChatApp;
