// react-chat-app/lib/cjs/Conversation.js
'use strict';

var React = require('react');
var styledComponents_esm = require('./styled-components.esm-92a8efa7.js');
var index = require('./index-e429b214.js');
require('./Bubble.js');
require('./_rollupPluginBabelHelpers-9bd7278f.js');
require('./theme-37c6f16c.js');
require('./Paragraph.js');
require('./style-0ac2425a.js');
require('./style-94d0e43c.js');
require('./index-4c52fc8e.js');
require('./Text.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

const Conversations = styledComponents_esm.Ye.div`
  padding: 10px 15px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  width: 100%;
  overflow-y: auto;
  flex: 1;
  & > * {
    margin: 10px 0;
  }
`;
const StyledConversation = styledComponents_esm.Ye.div`
  display: flex;
  flex-direction: column;
  height: 100vh;
  border: 1px solid #EFECE8;

  & > *:last-child {
    align-self: end;
  }
`;

function Conversation({
  titleBar,
  children,
  messageBar,
  ...rest
}) {
  const convsAnimeProps = index.useSpring({
    opacity: 1,
    from: {
      opacity: 0,
      transform: 'translate3d(50px, 0px, 0px)'
    },
    delay: 800
  });
  return /*#__PURE__*/React__default['default'].createElement(StyledConversation, rest, titleBar, /*#__PURE__*/React__default['default'].createElement(Conversations, {
    style: convsAnimeProps
  }, children), messageBar);
}

Conversation.propTypes = {
  children: styledComponents_esm.propTypes.any,
  titleBar: styledComponents_esm.propTypes.element,
  messageBar: styledComponents_esm.propTypes.element
};

module.exports = Conversation;
