// react-chat-app/lib/cjs/DropDown.js
'use strict';

var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-9bd7278f.js');
var React = require('react');
var styledComponents_esm = require('./styled-components.esm-92a8efa7.js');
var style = require('./style-b9a5dc7e.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

styledComponents_esm.Ye.div`
  margin: 12px 0;
`;
const DropdownContainer = styledComponents_esm.Ye.div`
  position: absolute;
  white-space: nowrap;
  padding: 4px 26px;
  background: white;
  box-shadow: 0px 4px 32px rgba(0, 0, 0, 0.08);

  display: ${({
  visible
}) => visible ? 'block' : 'none'};

  ${({
  align
}) => align}: 0;

  ${style.StyledDivider} {
    margin: -3px -26px;
    width: calc(100% + 52px);
  }
`;
const StyledDropdown = styledComponents_esm.Ye.div`
  position: relative;
  cursor: pointer;
`;

function Dropdown({
  children,
  content,
  align = 'right',
  ...rest
}) {
  const [visible, setVisible] = React.useState(false);
  return /*#__PURE__*/React__default['default'].createElement(StyledDropdown, _rollupPluginBabelHelpers._extends({
    onClick: () => setVisible(!visible)
  }, rest), children, content && /*#__PURE__*/React__default['default'].createElement(DropdownContainer, {
    align: align,
    visible: visible
  }, content));
}

Dropdown.propTypes = {
  children: styledComponents_esm.propTypes.any,
  content: styledComponents_esm.propTypes.any,
  align: styledComponents_esm.propTypes.oneOf(['top', 'left', 'bottom', 'right'])
};

module.exports = Dropdown;
