// react-chat-app/lib/cjs/FilterList.js
'use strict';

var React = require('react');
var styledComponents_esm = require('./styled-components.esm-92a8efa7.js');
var Input = require('./Input.js');
var Filter = require('./Filter.js');
var Select = require('./Select.js');
var Option = require('./Option.js');
var Button = require('./index-dc2aad31.js');
var Icon = require('./index-4c52fc8e.js');
require('./_rollupPluginBabelHelpers-9bd7278f.js');
require('./theme-37c6f16c.js');
require('./Text.js');
require('./style-94d0e43c.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

function _interopNamespace(e) {
  if (e && e.__esModule) return e;
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () {
            return e[k];
          }
        });
      }
    });
  }
  n['default'] = e;
  return Object.freeze(n);
}

var React__namespace = /*#__PURE__*/_interopNamespace(React);
var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

var _path;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function SvgPlus(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends({
    width: 12,
    height: 12,
    fill: "#fff",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path || (_path = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M10.351 6.923H6.923v3.428a.923.923 0 01-1.846 0V6.924H1.648A.92.92 0 01.726 6c0-.51.413-.923.923-.923h3.428V1.649a.923.923 0 011.846 0v3.428h3.428a.923.923 0 010 1.846z"
  })));
}

const StyledFilterList = styledComponents_esm.Ye.div`
  padding: 30px;
  height: 100vh;
  overflow-y: auto;
`;

function FilterList({
  children,
  sortOptions,
  filterLabel = 'Sort',
  actionLabel,
  ...rest
}) {
  return /*#__PURE__*/React__default['default'].createElement(StyledFilterList, rest, /*#__PURE__*/React__default['default'].createElement(Input.Search, null), /*#__PURE__*/React__default['default'].createElement(Filter, {
    style: {
      padding: '20px 0'
    }
  }, sortOptions && /*#__PURE__*/React__default['default'].createElement(Filter.Filters, {
    label: filterLabel
  }, /*#__PURE__*/React__default['default'].createElement(Select, null, sortOptions.map(_ => /*#__PURE__*/React__default['default'].createElement(Option, {
    key: _.key
  }, _.option)))), actionLabel && /*#__PURE__*/React__default['default'].createElement(Filter.Actions, {
    label: actionLabel
  }, /*#__PURE__*/React__default['default'].createElement(Button.Button, null, /*#__PURE__*/React__default['default'].createElement(Icon.Icon, {
    icon: SvgPlus,
    width: 12,
    height: 12
  })))), children);
}

FilterList.propTypes = {
  children: styledComponents_esm.propTypes.any,
  sortOptions: styledComponents_esm.propTypes.arrayOf(styledComponents_esm.propTypes.shape({
    key: styledComponents_esm.propTypes.string,
    option: styledComponents_esm.propTypes.string
  })),
  filterLabel: styledComponents_esm.propTypes.string,
  actionLabel: styledComponents_esm.propTypes.string
};

module.exports = FilterList;
