// react-chat-app/lib/cjs/index-c260b296.js
'use strict';

var React = require('react');
var styledComponents_esm = require('./styled-components.esm-92a8efa7.js');
var FilterList = require('./FilterList.js');
var mixins = require('./mixins-4c7023a3.js');
var Text = require('./Text.js');
var Badge = require('./Badge.js');
var style = require('./style-4ef8db8a.js');
var style$1 = require('./style-0ac2425a.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

const Title = styledComponents_esm.Ye(Text)`
  grid-area: title;
`;
const RightLabel = styledComponents_esm.Ye(Text)`
  grid-area: label;
  justify-self: end;
`;
const SubTitle = styledComponents_esm.Ye(Text)`
  grid-area: subtitle;
`;
const Description = styledComponents_esm.Ye.div`
  grid-area: description;
  display: grid;
  grid-template-columns: 1fr 30px;
  ${({
  iconSvg
}) => iconSvg && styledComponents_esm._e`
  grid-template-columns: 24px 1fr 30px;
  `}
  align-items: center;

`;
const UnreadBadge = styledComponents_esm.Ye(Badge)`
  justify-self: end;
`;
const StyledCard = styledComponents_esm.Ye.div`
  ${mixins.card()};
  display: grid;
  grid-template-areas:
    "avatar title label"
    "avatar subtitle sublabel"
    "description description description";
  grid-template-columns: 64px 1fr 1fr;
  row-gap: 16px;
  background: ${({
  background
}) => background};
  transition: 0.4s;
  &:hover {
    box-shadow: 0px 20px 50px rgba(0, 0, 0, 0.1);
  }

  ${style.AvatarWrapper} {
    grid-area: avatar;
  }

  ${({
  active
}) => active && styledComponents_esm._e`
      background: ${({
  theme
}) => theme.darkPurple};
      ${Title}, ${SubTitle}, ${RightLabel}, ${style$1.StyledParagraph} {
        color: white;
      }
      ${SubTitle}, ${RightLabel} {
        opacity: 0.4;
      }
      ${mixins.activeBar({
  barWidth: '4px',
  shadowWidth: '14px'
})}
      overflow: hidden;`}
`;

const ChatList = styledComponents_esm.Ye.div`
  ${StyledCard} {
    margin-bottom: 20px;
  }
`;
const StyledMessageList = styledComponents_esm.Ye.div``;

function MessageList({
  children,
  ...rest
}) {
  return /*#__PURE__*/React__default['default'].createElement(StyledMessageList, rest, /*#__PURE__*/React__default['default'].createElement(FilterList, null, /*#__PURE__*/React__default['default'].createElement(ChatList, null, children)));
}

MessageList.propTypes = {
  children: styledComponents_esm.propTypes.any
};

exports.Description = Description;
exports.MessageList = MessageList;
exports.RightLabel = RightLabel;
exports.StyledCard = StyledCard;
exports.SubTitle = SubTitle;
exports.Title = Title;
exports.UnreadBadge = UnreadBadge;
