// react-chat-app/lib/cjs/index-c4919884.js
'use strict';

var React = require('react');
var styledComponents_esm = require('./styled-components.esm-220693da.js');
var theme = require('./theme-37c6f16c.js');
var mixins = require('./mixins-379da3cd.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

const circleMixin = (color, size = '8px') => styledComponents_esm._e`
  content: "";
  display: block;
  position: absolute;
  ${mixins.circle(color, size)}
`;

const AvatarWrapper = styledComponents_esm.Ye.div`
  position: relative;
`;
const StatusIcon = styledComponents_esm.Ye.div`
  position: absolute;
  left: 2px;
  top: 4px;

  &::before {
    ${circleMixin('white')};
    transform: scale(2)
  }

  &::after {
    ${({
  theme,
  status,
  size
}) => {
  if (status === 'online') {
    return circleMixin(theme.green, size);
  }

  return circleMixin(theme.gray, size);
}}
  }
`;
const AvatarClip = styledComponents_esm.Ye.div`
    width: ${({
  size
}) => size};
    height: ${({
  size
}) => size};
    border-radius: 50%;
    overflow: hidden;
`;
const AvatarImage = styledComponents_esm.Ye.img`
    width: 100%;
    height: 100%;
    object-position: center
`;

function Avatar({
  src,
  size = '48px',
  status,
  StatusIconSize = '8px',
  ...rest
}) {
  return /*#__PURE__*/React__default['default'].createElement(styledComponents_esm.Le, {
    theme: theme.theme
  }, /*#__PURE__*/React__default['default'].createElement(AvatarWrapper, rest, status && /*#__PURE__*/React__default['default'].createElement(StatusIcon, {
    status: status,
    size: StatusIconSize
  }), /*#__PURE__*/React__default['default'].createElement(AvatarClip, {
    size: size
  }, /*#__PURE__*/React__default['default'].createElement(AvatarImage, {
    src: src,
    alt: ""
  }))));
}

Avatar.propTypes = {
  src: styledComponents_esm.propTypes.string,
  size: styledComponents_esm.propTypes.string,
  status: styledComponents_esm.propTypes.oneOf(['online', 'offline']),
  StatusIconSize: styledComponents_esm.propTypes.string
};
Avatar.defaultProps = {
  size: '48px',
  status: 'online',
  StatusIconSize: '8px'
};

exports.Avatar = Avatar;
exports.AvatarWrapper = AvatarWrapper;
