// react-chat-app/lib/cjs/index-4c52fc8e.js
'use strict';

var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-9bd7278f.js');
var React = require('react');
var styledComponents_esm = require('./styled-components.esm-92a8efa7.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

const IconWrapper = styledComponents_esm.Ye.div`
  display: inline-flex;
  align-items: center;
  justify-content: center;

  svg, 
  svg * {
    ${({
  color
}) => color ? `fill: ${color}` : ''};
    ${({
  opacity
}) => opacity ? `opcaity: ${opacity}` : ''};
  }

`;

function Icon({
  icon: IconSvg,
  width = 24,
  height = 24,
  opacity,
  color,
  ...rest
}) {
  return /*#__PURE__*/React__default['default'].createElement(IconWrapper, _rollupPluginBabelHelpers._extends({
    color: color,
    opacity: opacity
  }, rest), IconSvg && /*#__PURE__*/React__default['default'].createElement(IconSvg, {
    width: width,
    height: height
  }));
}

Icon.propTypes = {
  icon: styledComponents_esm.propTypes.oneOfType([styledComponents_esm.propTypes.func, styledComponents_esm.propTypes.object]),
  width: styledComponents_esm.propTypes.oneOfType([styledComponents_esm.propTypes.number, styledComponents_esm.propTypes.string]),
  height: styledComponents_esm.propTypes.oneOfType([styledComponents_esm.propTypes.number, styledComponents_esm.propTypes.string]),
  opacity: styledComponents_esm.propTypes.number,
  color: styledComponents_esm.propTypes.string
};
Icon.defaultProps = {
  width: 24,
  height: 24
};

exports.Icon = Icon;
exports.IconWrapper = IconWrapper;
