// react-chat-app/lib/cjs/Input.js
'use strict';

var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-9bd7278f.js');
var React = require('react');
var styledComponents_esm = require('./styled-components.esm-92a8efa7.js');
var Icon = require('./index-4c52fc8e.js');
var theme = require('./theme-37c6f16c.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

function _interopNamespace(e) {
  if (e && e.__esModule) return e;
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () {
            return e[k];
          }
        });
      }
    });
  }
  n['default'] = e;
  return Object.freeze(n);
}

var React__namespace = /*#__PURE__*/_interopNamespace(React);
var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

var _path;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function SvgSearch(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends({
    viewBox: "0 0 18 18",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path || (_path = /*#__PURE__*/React__namespace.createElement("path", {
    opacity: 0.3,
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M11.832 3.89a5.614 5.614 0 00-7.941 0 5.614 5.614 0 000 7.942 5.614 5.614 0 007.94 0 5.614 5.614 0 000-7.941zm5.84 13.782c-.44.438-1.15.438-1.59 0l-3.536-3.538c-3.08 2.302-7.446 2.084-10.243-.714a7.86 7.86 0 010-11.117 7.86 7.86 0 0111.117 0c2.798 2.797 3.016 7.164.714 10.243l3.538 3.538a1.124 1.124 0 010 1.588z",
    fill: "#292F4C"
  })));
}

const StyledInput = styledComponents_esm.Ye.input`
  width: 100%;
  height: 48px;
  border: none;
  background: none;
  color: ${({
  theme
}) => theme.grayDark};
  font-size: ${({
  theme
}) => theme.medium};
  display: block;
  &::placeholder {
    color: ${({
  theme
}) => theme.gray3};
  }
`;
const Prefix = styledComponents_esm.Ye.div`
  margin-right: 16px;
`;
const Suffix = styledComponents_esm.Ye.div`
  margin-left: 16px;
`;
const InputContainer = styledComponents_esm.Ye.div`
  display: flex;
  align-items: center;
  background: ${({
  theme
}) => theme.gray2};
  border-radius: 24px;
  padding: 0 30px;
`;

function Input({
  placeholder = 'Please input...',
  prefix,
  suffix,
  ...rest
}) {
  return /*#__PURE__*/React__default['default'].createElement(styledComponents_esm.Le, {
    theme: theme.theme
  }, /*#__PURE__*/React__default['default'].createElement(InputContainer, rest, prefix && /*#__PURE__*/React__default['default'].createElement(Prefix, null, prefix), /*#__PURE__*/React__default['default'].createElement(StyledInput, {
    placeholder: placeholder
  }), suffix && /*#__PURE__*/React__default['default'].createElement(Suffix, null, suffix)));
}

function Search({
  placeholder = 'Please input...',
  ...rest
}) {
  return /*#__PURE__*/React__default['default'].createElement(Input, _rollupPluginBabelHelpers._extends({
    placeholder: placeholder,
    prefix: /*#__PURE__*/React__default['default'].createElement(Icon.Icon, {
      icon: SvgSearch,
      color: "rgba(24, 28, 47, 0.3)",
      width: 18,
      height: 18
    })
  }, rest));
}

Input.Search = Search;
Input.propTypes = {
  placeholder: styledComponents_esm.propTypes.string,
  prefix: styledComponents_esm.propTypes.any,
  suffix: styledComponents_esm.propTypes.any
};
Search.propTypes = {
  placeholder: styledComponents_esm.propTypes.string,
  prefix: styledComponents_esm.propTypes.any,
  suffix: styledComponents_esm.propTypes.any
};

module.exports = Input;
