// react-chat-app/lib/cjs/MessageBar.js
'use strict';

var React = require('react');
var styledComponents_esm = require('./styled-components.esm-92a8efa7.js');
var Input = require('./Input.js');
var Icon = require('./index-4c52fc8e.js');
var Popover = require('./Popover.js');
var Button = require('./index-dc2aad31.js');
require('./_rollupPluginBabelHelpers-9bd7278f.js');
require('./theme-37c6f16c.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

function _interopNamespace(e) {
  if (e && e.__esModule) return e;
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () {
            return e[k];
          }
        });
      }
    });
  }
  n['default'] = e;
  return Object.freeze(n);
}

var React__namespace = /*#__PURE__*/_interopNamespace(React);
var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

var _path$3;

function _extends$3() { _extends$3 = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$3.apply(this, arguments); }

function SvgClip(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$3({
    viewBox: "0 0 22 20",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$3 || (_path$3 = /*#__PURE__*/React__namespace.createElement("path", {
    opacity: 0.3,
    d: "M1.23 18.745a4.156 4.156 0 005.953 0l9.523-9.707c.478-.483.743-1.13.744-1.82.004-1.379-1.178-2.584-2.53-2.58a2.482 2.482 0 00-1.788.762l-8.33 8.491a.87.87 0 000 1.214.83.83 0 001.19 0l8.332-8.495a.833.833 0 01.6-.256c.45 0 .843.4.843.86a.86.86 0 01-.25.607l-9.525 9.71a2.49 2.49 0 01-3.571 0 2.612 2.612 0 010-3.64L13.134 2.97a4.157 4.157 0 015.952 0c1.64 1.673 1.64 4.395 0 6.068l-8.333 8.494a.87.87 0 000 1.213.83.83 0 001.19 0l8.333-8.494c2.298-2.342 2.298-6.153 0-8.494a5.82 5.82 0 00-8.332 0L1.23 12.678c-1.642 1.673-1.642 4.394 0 6.067z",
    fill: "#292F4C"
  })));
}

var _path$2;

function _extends$2() { _extends$2 = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$2.apply(this, arguments); }

function SvgSmile(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$2({
    viewBox: "0 0 20 20",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$2 || (_path$2 = /*#__PURE__*/React__namespace.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M0 10C0 4.5 4.5 0 10 0s10 4.5 10 10-4.5 10-10 10S0 15.5 0 10zm15-2.5c0-.8-.7-1.5-1.5-1.5S12 6.7 12 7.5 12.7 9 13.5 9 15 8.3 15 7.5zM6.5 6C7.3 6 8 6.7 8 7.5S7.3 9 6.5 9 5 8.3 5 7.5 5.7 6 6.5 6zm-2.1 6c.8 2.3 3 4 5.6 4 2.6 0 4.8-1.7 5.6-4H4.4z",
    fill: "#4F9DDE"
  })));
}

var _path$1;

function _extends$1() { _extends$1 = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$1.apply(this, arguments); }

function SvgMicrophone(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$1({
    width: 15,
    height: 24,
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$1 || (_path$1 = /*#__PURE__*/React__namespace.createElement("path", {
    opacity: 0.3,
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M3.203 10.959V3.793c0-1.346.797-2.522 2-3.195v3.005c0 .354.326.646.73.646.406 0 .731-.291.731-.646V.074C6.935.03 7.213 0 7.5 0c.285 0 .563.03.834.074v3.529c0 .354.327.645.733.645.404 0 .732-.29.732-.645V.598c1.198.673 2 1.85 2 3.195v7.166c0 2.097-1.927 3.796-4.3 3.796-2.373 0-4.295-1.7-4.295-3.796zM.565 22.723h1l4.47-1.345v-4.252c-3.227-.597-5.66-3.13-5.66-6.158 0-.403.37-.73.827-.73.12 0 .238.025.34.065.29.116.492.369.492.665 0 2.663 2.45 4.827 5.465 4.827 3.013 0 5.467-2.164 5.467-4.827 0-.296.201-.549.488-.665a.94.94 0 01.34-.065c.46 0 .831.327.831.73 0 3.027-2.433 5.56-5.663 6.158v4.252l4.473 1.345h1c.313 0 .564.225.564.5v.276c0 .277-.25.501-.564.501H.565c-.313 0-.566-.224-.566-.5v-.277c0-.275.253-.5.566-.5z",
    fill: "#292F4C"
  })));
}

var _path;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function SvgPlane(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends({
    width: 24,
    height: 22,
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path || (_path = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M23.493.473a.354.354 0 00-.377-.053L.466 10.756a.343.343 0 00-.201.307.342.342 0 00.195.312l6.41 3.09a.354.354 0 00.362-.034l6.232-4.575L8.572 14.8a.34.34 0 00-.098.266l.487 6.23a.344.344 0 00.348.317c.1 0 .197-.042.264-.119l3.402-3.872 4.206 1.974a.355.355 0 00.29.005.345.345 0 00.193-.212L23.596.833a.34.34 0 00-.103-.36z",
    fill: "#fff"
  })));
}

const IconContainer = styledComponents_esm.Ye.div`
  display: flex;
  align-items: center;
  margin-right: -30px;
  & > * {
    margin-left: 16px;
  }
`;
const StyledPopoverContent = styledComponents_esm.Ye.div`
  display: flex;
  & > * {
    display: flex;
    margin: 0 8px;
    font-size: 16px;
  }
`;
const styledMessageBar = styledComponents_esm.Ye.footer`
  padding: 12px 30px;
  width: 100%;
`;

function DefaultButton() {
  return /*#__PURE__*/React__default['default'].createElement(Button.Button, {
    size: 52
  }, /*#__PURE__*/React__default['default'].createElement(Icon.Icon, {
    icon: SvgPlane,
    color: "white",
    style: {
      transform: 'translateX(-2px)'
    }
  }));
}

function MessageBar({
  children,
  prefixIcon = SvgClip,
  icon2 = SvgSmile,
  icon3 = SvgMicrophone,
  button: MessageButton = DefaultButton,
  popoverContent: PopoverContent,
  ...rest
}) {
  const [emojiIconActive, setEmojiIconActive] = React.useState(false);
  return /*#__PURE__*/React__default['default'].createElement(styledMessageBar, rest, /*#__PURE__*/React__default['default'].createElement(Input, {
    placeholder: "Please input....",
    prefix: /*#__PURE__*/React__default['default'].createElement(Icon.Icon, {
      icon: prefixIcon
    }),
    suffix: /*#__PURE__*/React__default['default'].createElement(IconContainer, null, /*#__PURE__*/React__default['default'].createElement(Popover, {
      content: PopoverContent && /*#__PURE__*/React__default['default'].createElement(StyledPopoverContent, null, /*#__PURE__*/React__default['default'].createElement(PopoverContent, null)),
      offset: {
        x: '-25%'
      },
      onVisible: () => setEmojiIconActive(true),
      onHide: () => setEmojiIconActive(false)
    }, /*#__PURE__*/React__default['default'].createElement(Icon.Icon, {
      icon: icon2,
      color: emojiIconActive ? undefined : 'rgba(24, 28, 47, 0.3)'
    })), /*#__PURE__*/React__default['default'].createElement(Icon.Icon, {
      icon: icon3
    }), MessageButton && /*#__PURE__*/React__default['default'].createElement(MessageButton, null))
  }));
}

MessageBar.propTypes = {
  children: styledComponents_esm.propTypes.any,
  button: styledComponents_esm.propTypes.any,
  icon2: styledComponents_esm.propTypes.any,
  prefixIcon: styledComponents_esm.propTypes.any,
  icon3: styledComponents_esm.propTypes.any,
  popoverContent: styledComponents_esm.propTypes.any
};

module.exports = MessageBar;
