// react-chat-app/lib/cjs/Navbar.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-9bd7278f.js');
var React = require('react');
var styledComponents_esm = require('./styled-components.esm-92a8efa7.js');
var Badge = require('./Badge.js');
var Icon = require('./index-4c52fc8e.js');
var mixins = require('./mixins-4c7023a3.js');
require('./theme-37c6f16c.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

const StyledNavBar = styledComponents_esm.Ye.div`
  height: 100vh;
`;
const StyledMenuItem = styledComponents_esm.Ye.div`
    & > a {
        width: 100%;
        height: 74px;
        display: flex;
        align-items: center;
        justify-content: center;
        ${mixins.activeBar()}
    }
`;

function NavBar({
  children,
  ...rest
}) {
  return /*#__PURE__*/React__default['default'].createElement(StyledNavBar, rest, children);
}

function MenuItem({
  icon,
  active = true,
  showBadge,
  ...rest
}) {
  return /*#__PURE__*/React__default['default'].createElement(StyledMenuItem, _rollupPluginBabelHelpers._extends({
    active: active
  }, rest), /*#__PURE__*/React__default['default'].createElement("a", {
    href: "#"
  }, /*#__PURE__*/React__default['default'].createElement(Badge, {
    show: showBadge
  }, icon && /*#__PURE__*/React__default['default'].createElement(Icon.Icon, {
    icon: icon
  }))));
}

NavBar.propTypes = {
  children: styledComponents_esm.propTypes.any
};
MenuItem.propTypes = {
  icon: styledComponents_esm.propTypes.any,
  active: styledComponents_esm.propTypes.bool,
  showBadge: styledComponents_esm.propTypes.bool
};

exports.MenuItem = MenuItem;
exports.default = NavBar;
