// react-chat-app/lib/cjs/Popover.js
'use strict';

var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-9bd7278f.js');
var React = require('react');
var styledComponents_esm = require('./styled-components.esm-92a8efa7.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

const StyledPopover = styledComponents_esm.Ye.div`
    position: relative;
    display: flex;
    justify-content: center;
`;
const Content = styledComponents_esm.Ye.div`
    background-color: ${({
  theme
}) => theme.background};
    border-radius: 21px;
    box-shadow: 0px 8px 40px rgba(0, 0, 0, 0.12);
    padding: 12px 30px;
    position: absolute;
    bottom: calc(100% + 16px);
    display: ${({
  visible
}) => visible ? 'block' : 'none'};
    ${({
  offset
}) => offset && `transform: translate(${offset.x || 0}, ${offset.y || 0})`};
`;
const Triangle = styledComponents_esm.Ye.div`
    position: absolute;
    width: 0;
    height: 0;
    border-style: solid;
    border-width: 6px 6px 0 6px;
    border-color: ${({
  theme
}) => theme.background} transparent transparent
    transparent;
    left: calc(50% - 6px);
    bottom: calc(100% + 12px - 5px);
    display: ${({
  visible
}) => visible ? 'block' : 'none'};
    ${({
  offset
}) => offset && `transform: translateY(${offset.y || 0});`}
`;
const Target = styledComponents_esm.Ye.div``;

function Popover({
  children,
  content,
  onHide,
  onVisible,
  offset,
  ...rest
}) {
  const [visible, setVisible] = React.useState(false);

  const onClick = () => {
    if (visible) {
      setVisible(false);

      if (onHide) {
        onHide();
      }
    } else {
      setVisible(true);

      if (onVisible) {
        onVisible();
      }
    }
  };

  return /*#__PURE__*/React__default['default'].createElement(StyledPopover, _rollupPluginBabelHelpers._extends({}, rest, {
    onClick: onClick
  }), /*#__PURE__*/React__default['default'].createElement(Content, {
    visible: visible
  }, content), /*#__PURE__*/React__default['default'].createElement(Triangle, {
    visible: visible
  }), /*#__PURE__*/React__default['default'].createElement(Target, null, children));
}

Popover.propTypes = {
  children: styledComponents_esm.propTypes.any,
  content: styledComponents_esm.propTypes.any,
  offset: styledComponents_esm.propTypes.shape({
    x: styledComponents_esm.propTypes.string,
    y: styledComponents_esm.propTypes.string
  }),
  onVisible: styledComponents_esm.propTypes.func,
  onHide: styledComponents_esm.propTypes.func
};

module.exports = Popover;
