// react-chat-app/lib/cjs/style-176b4b9e.js
'use strict';

var styledComponents_esm = require('./styled-components.esm-220693da.js');
var mixins = require('./mixins-379da3cd.js');

const circleMixin = (color, size = '8px') => styledComponents_esm._e`
  content: "";
  display: block;
  position: absolute;
  ${mixins.circle(color, size)}
`;

const AvatarWrapper = styledComponents_esm.Ye.div`
  position: relative;
`;
const StatusIcon = styledComponents_esm.Ye.div`
  position: absolute;
  left: 2px;
  top: 4px;

  &::before {
    ${circleMixin('white')};
    transform: scale(2)
  }

  &::after {
    ${({
  theme,
  status,
  size
}) => {
  if (status === 'online') {
    return circleMixin(theme.green, size);
  }

  return circleMixin(theme.gray, size);
}}
  }
`;
const AvatarClip = styledComponents_esm.Ye.div`
    width: ${({
  size
}) => size};
    height: ${({
  size
}) => size};
    border-radius: 50%;
    overflow: hidden;
`;
const AvatarImage = styledComponents_esm.Ye.img`
    width: 100%;
    height: 100%;
    object-position: center
`;

exports.AvatarClip = AvatarClip;
exports.AvatarImage = AvatarImage;
exports.AvatarWrapper = AvatarWrapper;
exports.StatusIcon = StatusIcon;
