// react-chat-app/lib/cjs/style-6ee798c9.js
'use strict';

var styledComponents_esm = require('./styled-components.esm-220693da.js');

const typeVariants = {
  primary: styledComponents_esm._e`
    color: ${({
    theme
  }) => theme.grayDark}
    `,
  secondary: styledComponents_esm._e`
    color: ${({
    theme
  }) => theme.grayDark};
    opacity: 0.3;
  `,
  danger: styledComponents_esm._e`
    color: ${({
    theme
  }) => theme.red};
  `
};
const sizeVariants = {
  normal: styledComponents_esm._e`
      ${({
    theme
  }) => theme.normal}
    `,
  medium: styledComponents_esm._e`
      ${({
    theme
  }) => theme.medium}
    `,
  large: styledComponents_esm._e`
      ${({
    theme
  }) => theme.large}
    `,
  xlarge: styledComponents_esm._e`
      ${({
    theme
  }) => theme.xlarge}
    `,
  xxlarge: styledComponents_esm._e`
      ${({
    theme
  }) => theme.xxlarge}
    `,
  small: styledComponents_esm._e`
      ${({
    theme
  }) => theme.small}
    `,
  xsmall: styledComponents_esm._e`
      ${({
    theme
  }) => theme.xsmall}
    `,
  xxsmall: styledComponents_esm._e`
      ${({
    theme
  }) => theme.xxsmall}
    `
};
const StyledText = styledComponents_esm.Ye.span`
    font-size: ${({
  size
}) => sizeVariants[size] || sizeVariants.normal};
    ${({
  type
}) => typeVariants[type]};
    ${({
  bold
}) => bold && 'font-weight: 500'};
  `;

exports.StyledText = StyledText;
