// react-chat-app/lib/cjs/Text.js
'use strict';

var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-9bd7278f.js');
var React = require('react');
var styledComponents_esm = require('./styled-components.esm-92a8efa7.js');
var theme = require('./theme-37c6f16c.js');
var style = require('./style-94d0e43c.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

function Text({
  children,
  type = 'primary',
  size = 'normal',
  bold,
  ...rest
}) {
  return /*#__PURE__*/React__default['default'].createElement(styledComponents_esm.Le, {
    theme: theme.theme
  }, /*#__PURE__*/React__default['default'].createElement(style.StyledText, _rollupPluginBabelHelpers._extends({
    type: type,
    size: size,
    bold: bold
  }, rest), children));
}

Text.propTypes = {
  children: styledComponents_esm.propTypes.any,
  type: styledComponents_esm.propTypes.oneOf(['primary', 'secondary', 'danger']),
  size: styledComponents_esm.propTypes.oneOf(['xxsmall', 'xsmall', 'small', 'normal', 'medium', 'large', 'xlarge', 'xxlarge']),
  bold: styledComponents_esm.propTypes.bool
};

module.exports = Text;
