// react-chat-app/lib/cjs/TitleBar.js
'use strict';

var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-9bd7278f.js');
var React = require('react');
var index = require('./index-e429b214.js');
var styledComponents_esm = require('./styled-components.esm-92a8efa7.js');
var Avatar = require('./Avatar.js');
var Paragraph = require('./Paragraph.js');
var Text = require('./Text.js');
var Icon = require('./index-4c52fc8e.js');
require('./theme-37c6f16c.js');
require('./style-4ef8db8a.js');
require('./mixins-4c7023a3.js');
require('./style-0ac2425a.js');
require('./style-94d0e43c.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

function _interopNamespace(e) {
  if (e && e.__esModule) return e;
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () {
            return e[k];
          }
        });
      }
    });
  }
  n['default'] = e;
  return Object.freeze(n);
}

var React__namespace = /*#__PURE__*/_interopNamespace(React);
var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

var _path$1;

function _extends$1() { _extends$1 = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$1.apply(this, arguments); }

function SvgCall(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$1({
    width: 17,
    height: 16,
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$1 || (_path$1 = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M16.293 12.754l-.081-.24c-.193-.561-.826-1.146-1.407-1.3l-2.15-.574c-.583-.155-1.415.053-1.841.47l-.779.761c-2.828-.747-5.046-2.916-5.809-5.68l.778-.761c.427-.417.64-1.23.482-1.8l-.587-2.103C4.741.957 4.142.339 3.57.153L3.324.072c-.574-.187-1.392.002-1.818.42L.342 1.63C.134 1.833 0 2.411 0 2.413a13.29 13.29 0 004.023 9.652c2.608 2.55 6.155 3.965 9.84 3.934.02 0 .628-.127.836-.33l1.164-1.138c.427-.417.62-1.217.429-1.777z",
    fill: "#292F4C"
  })));
}

var _path;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function SvgCamera(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends({
    width: 24,
    height: 14,
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path || (_path = /*#__PURE__*/React__namespace.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M2.626 0h10.95A2.607 2.607 0 0116.2 2.607v8.786c0 1.434-1.181 2.607-2.626 2.607H2.626C1.183 14 0 12.827 0 11.393V2.607C0 1.173 1.182 0 2.626 0zM21.61 1.616c.132-.078.29-.156.447-.182C23.08 1.199 24 1.98 24 2.946v8.056c0 .287-.079.547-.21.782a1.567 1.567 0 01-2.153.573l-4.123-2.346V3.963l4.096-2.347z",
    fill: "#292F4C"
  })));
}

const Title = styledComponents_esm.Ye.div`
  display: grid;
`;
const Actions = styledComponents_esm.Ye.div`
  display: flex;
  justify-content: space-between;
  align-items: center;

  ${Icon.IconWrapper} {
    cursor: pointer;
  }
`;
const StyledTitleBar = styledComponents_esm.Ye(index.animated.div)`
  display: grid;
  grid-template-columns: 62px 1fr 112px;
  padding: 30px;
  max-height: 110px;
  border-bottom: 1px solid #EFECE8;
`;

function TitleBar({
  animeProps = index.useSpring({
    opacity: 1,
    transform: 'translate3d(0px, 0px, 0px)',
    from: {
      opacity: 0,
      transform: 'translate3d(0px, -50px, 0px)'
    },
    delay: 500
  }),
  style,
  status = 'online',
  onAvatarClick,
  onVideoClicked,
  avatarUrl,
  dropDown: DropDown,
  children,
  ...rest
}) {
  return /*#__PURE__*/React__default['default'].createElement(StyledTitleBar, _rollupPluginBabelHelpers._extends({
    style: { ...style,
      ...animeProps
    }
  }, rest), /*#__PURE__*/React__default['default'].createElement(Avatar, {
    onClick: onAvatarClick,
    status: status,
    src: avatarUrl
  }), children, /*#__PURE__*/React__default['default'].createElement(Actions, null, /*#__PURE__*/React__default['default'].createElement(Icon.Icon, {
    opacity: 0.3,
    icon: SvgCall,
    onClick: onVideoClicked
  }), /*#__PURE__*/React__default['default'].createElement(Icon.Icon, {
    opacity: 0.3,
    icon: SvgCamera
  }), DropDown));
}

TitleBar.Title = ({
  name,
  title,
  subtitle
}) => /*#__PURE__*/React__default['default'].createElement(Title, null, /*#__PURE__*/React__default['default'].createElement(Paragraph, {
  size: "large"
}, name), /*#__PURE__*/React__default['default'].createElement(Paragraph, {
  type: "secondary"
}, /*#__PURE__*/React__default['default'].createElement(Text, null, title), /*#__PURE__*/React__default['default'].createElement(Text, null, ` ${subtitle}`)));

TitleBar.Title.propTypes = {
  name: styledComponents_esm.propTypes.string,
  title: styledComponents_esm.propTypes.string,
  subtitle: styledComponents_esm.propTypes.string
};
TitleBar.propTypes = {
  children: styledComponents_esm.propTypes.any,
  dropDown: styledComponents_esm.propTypes.any,
  onAvatarClick: styledComponents_esm.propTypes.func,
  status: styledComponents_esm.propTypes.oneOf(['online', 'offline']),
  onVideoClicked: styledComponents_esm.propTypes.func,
  animeProps: styledComponents_esm.propTypes.object,
  style: styledComponents_esm.propTypes.object,
  avatarUrl: styledComponents_esm.propTypes.string
};

module.exports = TitleBar;
