// react-chat-app/lib/cjs/VoiceMessage.js
'use strict';

var _rollupPluginBabelHelpers = require('./_rollupPluginBabelHelpers-9bd7278f.js');
var React = require('react');
var styledComponents_esm = require('./styled-components.esm-92a8efa7.js');
var Button = require('./index-dc2aad31.js');
var Icon = require('./index-4c52fc8e.js');
var Text = require('./Text.js');
var style = require('./style-94d0e43c.js');
require('./theme-37c6f16c.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

function _interopNamespace(e) {
  if (e && e.__esModule) return e;
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () {
            return e[k];
          }
        });
      }
    });
  }
  n['default'] = e;
  return Object.freeze(n);
}

var React__namespace = /*#__PURE__*/_interopNamespace(React);
var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

var _path$1;

function _extends$1() { _extends$1 = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$1.apply(this, arguments); }

function SvgPlay(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$1({
    viewBox: "0 0 14 16",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$1 || (_path$1 = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M12.685 6.956L2.764.777C2.471.605 2.107.633 1.831.633c-1.1 0-1.095.84-1.095 1.053v12.628c0 .18-.005 1.053 1.095 1.053.276 0 .64.028.933-.144l9.92-6.179C13.5 8.565 13.359 8 13.359 8s.141-.565-.673-1.044z",
    fill: "#fff"
  })));
}

var _path;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function SvgWave(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends({
    viewBox: "0 0 293 27",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path || (_path = /*#__PURE__*/React__namespace.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M244.999 1.5a1.5 1.5 0 013 0v24a1.5 1.5 0 11-3 0v-24zm5 3a1.5 1.5 0 013 0v19a1.5 1.5 0 11-3 0v-19zm26.5 5.5a1.5 1.5 0 00-1.5 1.5v4a1.5 1.5 0 003 0v-4a1.5 1.5 0 00-1.5-1.5zm13.5 3.5a1.5 1.5 0 113 0 1.5 1.5 0 01-3 0zm-5 0a1.5 1.5 0 113 0 1.5 1.5 0 01-3 0zm-28.5-7.5a1.5 1.5 0 00-1.5 1.5v12a1.5 1.5 0 003 0v-12a1.5 1.5 0 00-1.5-1.5zm3.5 2.5a1.5 1.5 0 013 0v11a1.5 1.5 0 11-3 0v-11zm6.5-.5a1.5 1.5 0 00-1.5 1.5v9a1.5 1.5 0 003 0v-9a1.5 1.5 0 00-1.5-1.5zm3.5 2.5a1.5 1.5 0 013 0v6a1.5 1.5 0 11-3 0v-6zm11.5.5a1.5 1.5 0 00-1.5 1.5v2a1.5 1.5 0 003 0v-2a1.5 1.5 0 00-1.5-1.5zm-40-4a1.5 1.5 0 00-1.5 1.5v10a1.5 1.5 0 003 0v-10a1.5 1.5 0 00-1.5-1.5zm-6.5.5a1.5 1.5 0 013 0v12a1.5 1.5 0 11-3 0v-12zm-3.5-1.5a1.5 1.5 0 00-1.5 1.5v12a1.5 1.5 0 003 0v-12a1.5 1.5 0 00-1.5-1.5zm-6.5 3.5a1.5 1.5 0 013 0v9a1.5 1.5 0 11-3 0v-9zm-3.5-2.5a1.5 1.5 0 00-1.5 1.5v10a1.5 1.5 0 003 0v-10a1.5 1.5 0 00-1.5-1.5zm-6.5-1.5a1.5 1.5 0 013 0v16a1.5 1.5 0 11-3 0v-16zm-3.5.5a1.5 1.5 0 00-1.5 1.5v12a1.5 1.5 0 003 0v-12a1.5 1.5 0 00-1.5-1.5zm-6.5 4.5a1.5 1.5 0 013 0v7a1.5 1.5 0 11-3 0v-7zm-3.5-3.5a1.5 1.5 0 00-1.5 1.5v10a1.5 1.5 0 003 0v-10a1.5 1.5 0 00-1.5-1.5zm-6.5.5a1.5 1.5 0 013 0v12a1.5 1.5 0 11-3 0v-12zm-3.5-.5a1.5 1.5 0 00-1.5 1.5v11a1.5 1.5 0 003 0v-11a1.5 1.5 0 00-1.5-1.5zm-6.5 2.5a1.5 1.5 0 013 0v9a1.5 1.5 0 11-3 0v-9zm-3.5-.5a1.5 1.5 0 00-1.5 1.5v6a1.5 1.5 0 003 0v-6a1.5 1.5 0 00-1.5-1.5zm-6.5-.5a1.5 1.5 0 013 0v11a1.5 1.5 0 11-3 0v-11zm-3.5-7.5a1.5 1.5 0 00-1.5 1.5v22a1.5 1.5 0 003 0v-22a1.5 1.5 0 00-1.5-1.5zm-6.5 9.5a1.5 1.5 0 013 0v6a1.5 1.5 0 11-3 0v-6zm-3.5-5.5a1.5 1.5 0 00-1.5 1.5v15a1.5 1.5 0 003 0v-15a1.5 1.5 0 00-1.5-1.5zm-93.5-.5a1.5 1.5 0 00-3 0v19a1.5 1.5 0 103 0v-19zm3.5.5A1.5 1.5 0 0173 6.5v15a1.5 1.5 0 01-3 0v-15A1.5 1.5 0 0171.5 5zM78 8.5a1.5 1.5 0 00-3 0v10a1.5 1.5 0 103 0v-10zM81.5 6A1.5 1.5 0 0183 7.5v12a1.5 1.5 0 01-3 0v-12A1.5 1.5 0 0181.5 6zM88 7.5a1.5 1.5 0 00-3 0v12a1.5 1.5 0 103 0v-12zM91.5 3A1.5 1.5 0 0193 4.5v18a1.5 1.5 0 01-3 0v-18A1.5 1.5 0 0191.5 3zM98 8.5a1.5 1.5 0 00-3 0v10a1.5 1.5 0 103 0v-10zm3.5-4.5a1.5 1.5 0 011.5 1.5v16a1.5 1.5 0 11-3 0v-16a1.5 1.5 0 011.5-1.5zm6.5-2.5a1.5 1.5 0 10-3 0v24a1.5 1.5 0 003 0v-24zm3.5 1.5a1.5 1.5 0 011.5 1.5v18a1.5 1.5 0 11-3 0v-18a1.5 1.5 0 011.5-1.5zm6.5 1.5a1.5 1.5 0 10-3 0v18a1.5 1.5 0 003 0v-18zm3.5 1.5a1.5 1.5 0 011.5 1.5v12a1.5 1.5 0 11-3 0v-12a1.5 1.5 0 011.5-1.5zm6.5 2.5a1.5 1.5 0 10-3 0v11a1.5 1.5 0 003 0v-11zm3.5-.5a1.5 1.5 0 011.5 1.5v9a1.5 1.5 0 11-3 0v-9a1.5 1.5 0 011.5-1.5zm6.5 2.5a1.5 1.5 0 10-3 0v6a1.5 1.5 0 003 0v-6zm3.5-3.5a1.5 1.5 0 011.5 1.5v11a1.5 1.5 0 11-3 0v-11a1.5 1.5 0 011.5-1.5zm6.5-1.5a1.5 1.5 0 10-3 0v16a1.5 1.5 0 003 0v-16zm3.5 3.5a1.5 1.5 0 011.5 1.5v6a1.5 1.5 0 11-3 0v-6a1.5 1.5 0 011.5-1.5zm6.5-2.5a1.5 1.5 0 10-3 0v15a1.5 1.5 0 003 0v-15zM61.499 3A1.5 1.5 0 0060 4.5v19a1.5 1.5 0 103 0v-19A1.5 1.5 0 0061.5 3zM55 1.5a1.5 1.5 0 113 0v24a1.5 1.5 0 01-3 0v-24zM51.5 7A1.5 1.5 0 0050 8.5v10a1.5 1.5 0 103 0v-10A1.5 1.5 0 0051.5 7zm-6.5.5a1.5 1.5 0 113 0v12a1.5 1.5 0 01-3 0v-12zM41.5 6A1.5 1.5 0 0040 7.5v12a1.5 1.5 0 103 0v-12A1.5 1.5 0 0041.5 6zM35 9.5a1.5 1.5 0 113 0v9a1.5 1.5 0 01-3 0v-9zM16.5 13a1.5 1.5 0 100 3 1.5 1.5 0 000-3zM10 14.5a1.5 1.5 0 113 0 1.5 1.5 0 01-3 0zM6.5 13a1.5 1.5 0 100 3 1.5 1.5 0 000-3zM0 14.5a1.5 1.5 0 113 0 1.5 1.5 0 01-3 0zM26.5 11a1.5 1.5 0 00-1.5 1.5v4a1.5 1.5 0 103 0v-4a1.5 1.5 0 00-1.5-1.5zm3.5.5a1.5 1.5 0 113 0v6a1.5 1.5 0 01-3 0v-6zm-8.5.5a1.5 1.5 0 00-1.5 1.5v2a1.5 1.5 0 103 0v-2a1.5 1.5 0 00-1.5-1.5z",
    fill: "#4F9DDE"
  })));
}

const typeVariants = {
  from: styledComponents_esm._e`
      ${Button.StyledButton} {
        background-color: white;
  
        ${Icon.IconWrapper} path {
          fill: ${({
    theme
  }) => theme.primaryColor};
        }
      }
      & > ${Icon.IconWrapper} path {
        fill: white;
      }
  
      & > ${style.StyledText} {
        color: white;
      }
    `
};
const StyledVoiceMessage = styledComponents_esm.Ye.div`

    display: flex;
    align-items: center;
    align-content: center;
    & > *:first-child {
    flex-shrink: 0;
    }

    & > *:not(:first-child) {
    margin-left: 16px;
    }

    ${({
  type
}) => type && typeVariants[type]}
`;

function VoiceMessage({
  children,
  type,
  time,
  ...rest
}) {
  return /*#__PURE__*/React__default['default'].createElement(StyledVoiceMessage, _rollupPluginBabelHelpers._extends({
    type: type
  }, rest), /*#__PURE__*/React__default['default'].createElement(Button.Button, {
    size: 40
  }, /*#__PURE__*/React__default['default'].createElement(Icon.Icon, {
    icon: SvgPlay,
    color: "white",
    width: "14",
    height: "16",
    style: {
      transform: 'translateX(1px)'
    }
  })), /*#__PURE__*/React__default['default'].createElement(Icon.Icon, {
    icon: SvgWave,
    width: "100%",
    height: "100%",
    color: "#4F9DDE"
  }), /*#__PURE__*/React__default['default'].createElement(Text, {
    bold: true
  }, time));
}

VoiceMessage.propTypes = {
  children: styledComponents_esm.propTypes.any,
  type: styledComponents_esm.propTypes.oneOf(['from']),
  time: styledComponents_esm.propTypes.string
};

module.exports = VoiceMessage;
