// react-chat-app/lib/esm/Avatar.js
import React__default from 'react';
import { L as Le, p as propTypes } from './styled-components.esm-a60371c1.js';
import { t as theme } from './theme-e593c2e7.js';
import { A as AvatarWrapper, S as StatusIcon, a as AvatarClip, b as AvatarImage } from './style-07254a25.js';
import './mixins-b7fe34bc.js';

function Avatar({
  src,
  size = '48px',
  status,
  StatusIconSize = '8px',
  ...rest
}) {
  return /*#__PURE__*/React__default.createElement(Le, {
    theme: theme
  }, /*#__PURE__*/React__default.createElement(AvatarWrapper, rest, status && /*#__PURE__*/React__default.createElement(StatusIcon, {
    status: status,
    size: StatusIconSize
  }), /*#__PURE__*/React__default.createElement(AvatarClip, {
    size: size
  }, /*#__PURE__*/React__default.createElement(AvatarImage, {
    src: src,
    alt: ""
  }))));
}

Avatar.propTypes = {
  src: propTypes.string,
  size: propTypes.string,
  status: propTypes.oneOf(['online', 'offline']),
  StatusIconSize: propTypes.string
};
Avatar.defaultProps = {
  size: '48px',
  status: 'online',
  StatusIconSize: '8px'
};

export default Avatar;
