// react-chat-app/lib/esm/Bubble.js
import { _ as _extends$1 } from './_rollupPluginBabelHelpers-1b8c042f.js';
import * as React from 'react';
import React__default from 'react';
import { Y as Ye, _ as _e, L as Le, p as propTypes } from './styled-components.esm-a60371c1.js';
import { t as theme } from './theme-e593c2e7.js';
import Paragraph from './Paragraph.js';
import { I as Icon } from './index-3c33a3b3.js';
import Text from './Text.js';
import './style-d40a0211.js';
import './style-fb8054ef.js';

var _path;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function SvgBubbleTip(props) {
  return /*#__PURE__*/React.createElement("svg", _extends({
    width: 29,
    height: 16,
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path || (_path = /*#__PURE__*/React.createElement("path", {
    d: "M8.488 1.146C8.165 2.53 6.728 6.538.511 10.919a.989.989 0 00.356 1.774c.034.008.867.19 2.364.32 4.266.374 13.06.29 24.214-4.426.82.116-23.067-10.41-18.957-7.441z",
    fill: "#fff"
  })));
}

const Time = Ye(Paragraph).attrs({
  type: 'secondary',
  size: 'small'
})`
  margin: 6px;
  margin-left: 24px;
  word-spacing: 1rem;
`;
const BubbleTip = Ye(Icon)`
  position: absolute;
  bottom: -15%;
  left: 0;
  z-index: 5;
`;
const BubbleWrapper = Ye.div`
  padding: 15px 22px;
  box-shadow: 0px 8px 24px rgba(0, 0, 0, 0.1);
  border-radius: 100px;
  position: relative;
  z-index: 10;
`;
const MessageText = Ye(Text)``;
const typeVariants = {
  from: _e`
    align-self: flex-end;
    ${BubbleWrapper} {
      background-color: ${({
    theme
  }) => theme.primaryColor};
    }
    ${BubbleTip} {
      transform: rotateY(180deg);
      left: unset;
      right: 0;
      /* svg color */
      path {
        fill: ${({
    theme
  }) => theme.primaryColor};
      }
    }
    ${Time} {
      text-align: right;
      margin-left: 0;
      margin-right: 24px;
    }
    ${MessageText} {
      color: white;
    }
  `
};
const StyledBubble = Ye.div`
  display: flex;
  flex-direction: column;
  ${({
  type
}) => type && typeVariants[type]}
`;

function Bubble({
  children,
  type,
  time,
  ...rest
}) {
  return /*#__PURE__*/React__default.createElement(Le, {
    theme: theme
  }, /*#__PURE__*/React__default.createElement(StyledBubble, _extends$1({
    type: type
  }, rest), /*#__PURE__*/React__default.createElement(BubbleWrapper, null, /*#__PURE__*/React__default.createElement(BubbleTip, {
    icon: SvgBubbleTip,
    width: 40,
    height: 28,
    color: "white"
  }), /*#__PURE__*/React__default.createElement(MessageText, null, children)), /*#__PURE__*/React__default.createElement(Time, null, time)));
}

Bubble.propTypes = {
  children: propTypes.any,
  type: propTypes.oneOf(['from', 'to']),
  time: propTypes.string
};

export default Bubble;
