// react-chat-app/lib/esm/ChatApp.js
import React__default from 'react';
import { Y as Ye, p as propTypes } from './styled-components.esm-a60371c1.js';
import { u as useSpring, a as animated } from './index-2aec7d0d.js';

const Nav = Ye.nav`
  flex-shrink: 0;

  position: relative;
  z-index: 100;
`;
const Sidebar = Ye.aside`
  max-width: 448px;
  min-width: 344px;
  height: 100vh;
  flex: 1;
  background: ${({
  theme
}) => theme.grediantGray};

  position: relative;
  z-index: 50;
  > div {
    will-change: transform, opacity;
    position: absolute;
    width: 100%;
  }
`;
const Content = Ye.main`
  flex: 2;
  position: relative;
`;
const StyledChatApp = Ye.div`
  display: flex;
  height: 100vh;
  width: 100vw;
  overflow: hidden;
  position: relative;
`;

function ChatApp({
  children,
  navBar: NavBar,
  sideBar: MessageList,
  content: Conversation,
  profile: Profile,
  showProfile = true,
  ...rest
}) {
  const animeProps = showProfile ? useSpring({
    opacity: 1,
    transform: 'translate3d(0px, 0px, 0px)',
    from: {
      opacity: 0,
      transform: 'translate3d(-100px, 0px, 0px)'
    },
    delay: 500
  }) : {
    width: 0
  };
  return /*#__PURE__*/React__default.createElement(StyledChatApp, rest, /*#__PURE__*/React__default.createElement(Nav, null, NavBar && /*#__PURE__*/React__default.createElement(NavBar, null)), /*#__PURE__*/React__default.createElement(Sidebar, null, MessageList && /*#__PURE__*/React__default.createElement(MessageList, null)), /*#__PURE__*/React__default.createElement(Content, null, Conversation && /*#__PURE__*/React__default.createElement(Conversation, null)), /*#__PURE__*/React__default.createElement(animated.div, {
    style: { ...animeProps
    }
  }, Profile && /*#__PURE__*/React__default.createElement(Profile, null)));
}

ChatApp.propTypes = {
  children: propTypes.any,
  navBar: propTypes.any,
  sideBar: propTypes.any,
  content: propTypes.any,
  profile: propTypes.any,
  showProfile: propTypes.bool
};

export default ChatApp;
