// react-chat-app/lib/esm/Conversation.js
import React__default from 'react';
import { Y as Ye, p as propTypes } from './styled-components.esm-a60371c1.js';
import { u as useSpring } from './index-2aec7d0d.js';
import './Bubble.js';
import './_rollupPluginBabelHelpers-1b8c042f.js';
import './theme-e593c2e7.js';
import './Paragraph.js';
import './style-d40a0211.js';
import './style-fb8054ef.js';
import './index-3c33a3b3.js';
import './Text.js';

const Conversations = Ye.div`
  padding: 10px 15px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  width: 100%;
  overflow-y: auto;
  flex: 1;
  & > * {
    margin: 10px 0;
  }
`;
const StyledConversation = Ye.div`
  display: flex;
  flex-direction: column;
  height: 100vh;
  border: 1px solid #EFECE8;

  & > *:last-child {
    align-self: end;
  }
`;

function Conversation({
  titleBar,
  children,
  messageBar,
  ...rest
}) {
  const convsAnimeProps = useSpring({
    opacity: 1,
    from: {
      opacity: 0,
      transform: 'translate3d(50px, 0px, 0px)'
    },
    delay: 800
  });
  return /*#__PURE__*/React__default.createElement(StyledConversation, rest, titleBar, /*#__PURE__*/React__default.createElement(Conversations, {
    style: convsAnimeProps
  }, children), messageBar);
}

Conversation.propTypes = {
  children: propTypes.any,
  titleBar: propTypes.element,
  messageBar: propTypes.element
};

export default Conversation;
