// react-chat-app/lib/esm/DropDown.js
import { _ as _extends } from './_rollupPluginBabelHelpers-1b8c042f.js';
import React__default, { useState } from 'react';
import { Y as Ye, p as propTypes } from './styled-components.esm-a60371c1.js';
import { S as StyledDivider } from './style-85ec59ca.js';

Ye.div`
  margin: 12px 0;
`;
const DropdownContainer = Ye.div`
  position: absolute;
  white-space: nowrap;
  padding: 4px 26px;
  background: white;
  box-shadow: 0px 4px 32px rgba(0, 0, 0, 0.08);

  display: ${({
  visible
}) => visible ? 'block' : 'none'};

  ${({
  align
}) => align}: 0;

  ${StyledDivider} {
    margin: -3px -26px;
    width: calc(100% + 52px);
  }
`;
const StyledDropdown = Ye.div`
  position: relative;
  cursor: pointer;
`;

function Dropdown({
  children,
  content,
  align = 'right',
  ...rest
}) {
  const [visible, setVisible] = useState(false);
  return /*#__PURE__*/React__default.createElement(StyledDropdown, _extends({
    onClick: () => setVisible(!visible)
  }, rest), children, content && /*#__PURE__*/React__default.createElement(DropdownContainer, {
    align: align,
    visible: visible
  }, content));
}

Dropdown.propTypes = {
  children: propTypes.any,
  content: propTypes.any,
  align: propTypes.oneOf(['top', 'left', 'bottom', 'right'])
};

export default Dropdown;
