// react-chat-app/lib/esm/FilterList.js
import * as React from 'react';
import React__default from 'react';
import { Y as Ye, p as propTypes } from './styled-components.esm-a60371c1.js';
import Input from './Input.js';
import Filter from './Filter.js';
import Select from './Select.js';
import Option from './Option.js';
import { B as Button } from './index-d7bca72b.js';
import { I as Icon } from './index-3c33a3b3.js';
import './_rollupPluginBabelHelpers-1b8c042f.js';
import './theme-e593c2e7.js';
import './Text.js';
import './style-fb8054ef.js';

var _path;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function SvgPlus(props) {
  return /*#__PURE__*/React.createElement("svg", _extends({
    width: 12,
    height: 12,
    fill: "#fff",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path || (_path = /*#__PURE__*/React.createElement("path", {
    d: "M10.351 6.923H6.923v3.428a.923.923 0 01-1.846 0V6.924H1.648A.92.92 0 01.726 6c0-.51.413-.923.923-.923h3.428V1.649a.923.923 0 011.846 0v3.428h3.428a.923.923 0 010 1.846z"
  })));
}

const StyledFilterList = Ye.div`
  padding: 30px;
  height: 100vh;
  overflow-y: auto;
`;

function FilterList({
  children,
  sortOptions,
  filterLabel = 'Sort',
  actionLabel,
  ...rest
}) {
  return /*#__PURE__*/React__default.createElement(StyledFilterList, rest, /*#__PURE__*/React__default.createElement(Input.Search, null), /*#__PURE__*/React__default.createElement(Filter, {
    style: {
      padding: '20px 0'
    }
  }, sortOptions && /*#__PURE__*/React__default.createElement(Filter.Filters, {
    label: filterLabel
  }, /*#__PURE__*/React__default.createElement(Select, null, sortOptions.map(_ => /*#__PURE__*/React__default.createElement(Option, {
    key: _.key
  }, _.option)))), actionLabel && /*#__PURE__*/React__default.createElement(Filter.Actions, {
    label: actionLabel
  }, /*#__PURE__*/React__default.createElement(Button, null, /*#__PURE__*/React__default.createElement(Icon, {
    icon: SvgPlus,
    width: 12,
    height: 12
  })))), children);
}

FilterList.propTypes = {
  children: propTypes.any,
  sortOptions: propTypes.arrayOf(propTypes.shape({
    key: propTypes.string,
    option: propTypes.string
  })),
  filterLabel: propTypes.string,
  actionLabel: propTypes.string
};

export default FilterList;
