// react-chat-app/lib/esm/index-3c33a3b3.js
import { _ as _extends } from './_rollupPluginBabelHelpers-1b8c042f.js';
import React__default from 'react';
import { Y as Ye, p as propTypes } from './styled-components.esm-a60371c1.js';

const IconWrapper = Ye.div`
  display: inline-flex;
  align-items: center;
  justify-content: center;

  svg, 
  svg * {
    ${({
  color
}) => color ? `fill: ${color}` : ''};
    ${({
  opacity
}) => opacity ? `opcaity: ${opacity}` : ''};
  }

`;

function Icon({
  icon: IconSvg,
  width = 24,
  height = 24,
  opacity,
  color,
  ...rest
}) {
  return /*#__PURE__*/React__default.createElement(IconWrapper, _extends({
    color: color,
    opacity: opacity
  }, rest), IconSvg && /*#__PURE__*/React__default.createElement(IconSvg, {
    width: width,
    height: height
  }));
}

Icon.propTypes = {
  icon: propTypes.oneOfType([propTypes.func, propTypes.object]),
  width: propTypes.oneOfType([propTypes.number, propTypes.string]),
  height: propTypes.oneOfType([propTypes.number, propTypes.string]),
  opacity: propTypes.number,
  color: propTypes.string
};
Icon.defaultProps = {
  width: 24,
  height: 24
};

export { Icon as I, IconWrapper as a };
