// react-chat-app/lib/esm/index-aa2c2b75.js
import React__default from 'react';
import { Y as Ye, _ as _e, L as Le, p as propTypes } from './styled-components.esm-e7249c67.js';
import { t as theme } from './theme-e593c2e7.js';
import { c as circle } from './mixins-dd0007f3.js';

const circleMixin = (color, size = '8px') => _e`
  content: "";
  display: block;
  position: absolute;
  ${circle(color, size)}
`;

const AvatarWrapper = Ye.div`
  position: relative;
`;
const StatusIcon = Ye.div`
  position: absolute;
  left: 2px;
  top: 4px;

  &::before {
    ${circleMixin('white')};
    transform: scale(2)
  }

  &::after {
    ${({
  theme,
  status,
  size
}) => {
  if (status === 'online') {
    return circleMixin(theme.green, size);
  }

  return circleMixin(theme.gray, size);
}}
  }
`;
const AvatarClip = Ye.div`
    width: ${({
  size
}) => size};
    height: ${({
  size
}) => size};
    border-radius: 50%;
    overflow: hidden;
`;
const AvatarImage = Ye.img`
    width: 100%;
    height: 100%;
    object-position: center
`;

function Avatar({
  src,
  size = '48px',
  status,
  StatusIconSize = '8px',
  ...rest
}) {
  return /*#__PURE__*/React__default.createElement(Le, {
    theme: theme
  }, /*#__PURE__*/React__default.createElement(AvatarWrapper, rest, status && /*#__PURE__*/React__default.createElement(StatusIcon, {
    status: status,
    size: StatusIconSize
  }), /*#__PURE__*/React__default.createElement(AvatarClip, {
    size: size
  }, /*#__PURE__*/React__default.createElement(AvatarImage, {
    src: src,
    alt: ""
  }))));
}

Avatar.propTypes = {
  src: propTypes.string,
  size: propTypes.string,
  status: propTypes.oneOf(['online', 'offline']),
  StatusIconSize: propTypes.string
};
Avatar.defaultProps = {
  size: '48px',
  status: 'online',
  StatusIconSize: '8px'
};

export { Avatar as A, AvatarWrapper as a };
