// react-chat-app/lib/esm/index-11bab19a.js
import React__default from 'react';
import { Y as Ye, _ as _e, p as propTypes } from './styled-components.esm-a60371c1.js';
import FilterList from './FilterList.js';
import { b as card, a as activeBar } from './mixins-b7fe34bc.js';
import Text from './Text.js';
import Badge from './Badge.js';
import { A as AvatarWrapper } from './style-07254a25.js';
import { S as StyledParagraph } from './style-d40a0211.js';

const Title = Ye(Text)`
  grid-area: title;
`;
const RightLabel = Ye(Text)`
  grid-area: label;
  justify-self: end;
`;
const SubTitle = Ye(Text)`
  grid-area: subtitle;
`;
const Description = Ye.div`
  grid-area: description;
  display: grid;
  grid-template-columns: 1fr 30px;
  ${({
  iconSvg
}) => iconSvg && _e`
  grid-template-columns: 24px 1fr 30px;
  `}
  align-items: center;

`;
const UnreadBadge = Ye(Badge)`
  justify-self: end;
`;
const StyledCard = Ye.div`
  ${card()};
  display: grid;
  grid-template-areas:
    "avatar title label"
    "avatar subtitle sublabel"
    "description description description";
  grid-template-columns: 64px 1fr 1fr;
  row-gap: 16px;
  background: ${({
  background
}) => background};
  transition: 0.4s;
  &:hover {
    box-shadow: 0px 20px 50px rgba(0, 0, 0, 0.1);
  }

  ${AvatarWrapper} {
    grid-area: avatar;
  }

  ${({
  active
}) => active && _e`
      background: ${({
  theme
}) => theme.darkPurple};
      ${Title}, ${SubTitle}, ${RightLabel}, ${StyledParagraph} {
        color: white;
      }
      ${SubTitle}, ${RightLabel} {
        opacity: 0.4;
      }
      ${activeBar({
  barWidth: '4px',
  shadowWidth: '14px'
})}
      overflow: hidden;`}
`;

const ChatList = Ye.div`
  ${StyledCard} {
    margin-bottom: 20px;
  }
`;
const StyledMessageList = Ye.div``;

function MessageList({
  children,
  ...rest
}) {
  return /*#__PURE__*/React__default.createElement(StyledMessageList, rest, /*#__PURE__*/React__default.createElement(FilterList, null, /*#__PURE__*/React__default.createElement(ChatList, null, children)));
}

MessageList.propTypes = {
  children: propTypes.any
};

export { Description as D, MessageList as M, RightLabel as R, StyledCard as S, Title as T, UnreadBadge as U, SubTitle as a };
