// react-chat-app/lib/esm/Input.js
import { _ as _extends$1 } from './_rollupPluginBabelHelpers-1b8c042f.js';
import * as React from 'react';
import React__default from 'react';
import { Y as Ye, L as Le, p as propTypes } from './styled-components.esm-a60371c1.js';
import { I as Icon } from './index-3c33a3b3.js';
import { t as theme } from './theme-e593c2e7.js';

var _path;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function SvgSearch(props) {
  return /*#__PURE__*/React.createElement("svg", _extends({
    viewBox: "0 0 18 18",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path || (_path = /*#__PURE__*/React.createElement("path", {
    opacity: 0.3,
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M11.832 3.89a5.614 5.614 0 00-7.941 0 5.614 5.614 0 000 7.942 5.614 5.614 0 007.94 0 5.614 5.614 0 000-7.941zm5.84 13.782c-.44.438-1.15.438-1.59 0l-3.536-3.538c-3.08 2.302-7.446 2.084-10.243-.714a7.86 7.86 0 010-11.117 7.86 7.86 0 0111.117 0c2.798 2.797 3.016 7.164.714 10.243l3.538 3.538a1.124 1.124 0 010 1.588z",
    fill: "#292F4C"
  })));
}

const StyledInput = Ye.input`
  width: 100%;
  height: 48px;
  border: none;
  background: none;
  color: ${({
  theme
}) => theme.grayDark};
  font-size: ${({
  theme
}) => theme.medium};
  display: block;
  &::placeholder {
    color: ${({
  theme
}) => theme.gray3};
  }
`;
const Prefix = Ye.div`
  margin-right: 16px;
`;
const Suffix = Ye.div`
  margin-left: 16px;
`;
const InputContainer = Ye.div`
  display: flex;
  align-items: center;
  background: ${({
  theme
}) => theme.gray2};
  border-radius: 24px;
  padding: 0 30px;
`;

function Input({
  placeholder = 'Please input...',
  prefix,
  suffix,
  ...rest
}) {
  return /*#__PURE__*/React__default.createElement(Le, {
    theme: theme
  }, /*#__PURE__*/React__default.createElement(InputContainer, rest, prefix && /*#__PURE__*/React__default.createElement(Prefix, null, prefix), /*#__PURE__*/React__default.createElement(StyledInput, {
    placeholder: placeholder
  }), suffix && /*#__PURE__*/React__default.createElement(Suffix, null, suffix)));
}

function Search({
  placeholder = 'Please input...',
  ...rest
}) {
  return /*#__PURE__*/React__default.createElement(Input, _extends$1({
    placeholder: placeholder,
    prefix: /*#__PURE__*/React__default.createElement(Icon, {
      icon: SvgSearch,
      color: "rgba(24, 28, 47, 0.3)",
      width: 18,
      height: 18
    })
  }, rest));
}

Input.Search = Search;
Input.propTypes = {
  placeholder: propTypes.string,
  prefix: propTypes.any,
  suffix: propTypes.any
};
Search.propTypes = {
  placeholder: propTypes.string,
  prefix: propTypes.any,
  suffix: propTypes.any
};

export default Input;
