// react-chat-app/lib/esm/Navbar.js
import { _ as _extends } from './_rollupPluginBabelHelpers-1b8c042f.js';
import React__default from 'react';
import { Y as Ye, p as propTypes } from './styled-components.esm-a60371c1.js';
import Badge from './Badge.js';
import { I as Icon } from './index-3c33a3b3.js';
import { a as activeBar } from './mixins-b7fe34bc.js';
import './theme-e593c2e7.js';

const StyledNavBar = Ye.div`
  height: 100vh;
`;
const StyledMenuItem = Ye.div`
    & > a {
        width: 100%;
        height: 74px;
        display: flex;
        align-items: center;
        justify-content: center;
        ${activeBar()}
    }
`;

function NavBar({
  children,
  ...rest
}) {
  return /*#__PURE__*/React__default.createElement(StyledNavBar, rest, children);
}

function MenuItem({
  icon,
  active = true,
  showBadge,
  ...rest
}) {
  return /*#__PURE__*/React__default.createElement(StyledMenuItem, _extends({
    active: active
  }, rest), /*#__PURE__*/React__default.createElement("a", {
    href: "#"
  }, /*#__PURE__*/React__default.createElement(Badge, {
    show: showBadge
  }, icon && /*#__PURE__*/React__default.createElement(Icon, {
    icon: icon
  }))));
}

NavBar.propTypes = {
  children: propTypes.any
};
MenuItem.propTypes = {
  icon: propTypes.any,
  active: propTypes.bool,
  showBadge: propTypes.bool
};

export default NavBar;
export { MenuItem };
