// react-chat-app/lib/esm/Profile.js
import { Y as Ye, p as propTypes } from './styled-components.esm-a60371c1.js';
import * as React from 'react';
import React__default from 'react';
import Avatar from './Avatar.js';
import Paragraph from './Paragraph.js';
import Divider from './Divider.js';
import Text from './Text.js';
import { B as Button } from './index-d7bca72b.js';
import { I as Icon } from './index-3c33a3b3.js';
import './theme-e593c2e7.js';
import './style-07254a25.js';
import './mixins-b7fe34bc.js';
import './_rollupPluginBabelHelpers-1b8c042f.js';
import './style-d40a0211.js';
import './style-fb8054ef.js';
import './style-85ec59ca.js';

var _path;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function SvgCross(props) {
  return /*#__PURE__*/React.createElement("svg", _extends({
    width: 20,
    height: 20,
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path || (_path = /*#__PURE__*/React.createElement("path", {
    opacity: 0.12,
    d: "M19.324 3.94L13.264 10l6.06 6.06a2.308 2.308 0 01-3.263 3.264L10 13.264l-6.06 6.06A2.3 2.3 0 012.307 20a2.308 2.308 0 01-1.631-3.94L6.736 10 .675 3.94A2.308 2.308 0 013.939.675l6.06 6.06 6.062-6.06a2.308 2.308 0 013.263 3.263z",
    fill: "#292F4C"
  })));
}

var arrowRight = "data:image/svg+xml,%3Csvg%20width%3D%225%22%20height%3D%228%22%20viewBox%3D%220%200%205%208%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%3Cpath%20d%3D%22M5%204.00001C5%204.14339%204.93998%204.28675%204.82018%204.39606L1.0486%207.83589C0.808678%208.0547%200.419689%208.0547%200.179867%207.83589C-0.0599556%207.61716%20-0.0599556%207.26245%200.179867%207.04362L3.51717%204.00001L0.179983%200.956392C-0.0598391%200.737575%20-0.0598391%200.382905%200.179983%200.164193C0.419806%20-0.0547314%200.808794%20-0.0547314%201.04871%200.164193L4.8203%203.60397C4.94011%203.71333%205%203.85669%205%204.00001Z%22%20fill%3D%22%234F9DDE%22%2F%3E%3C%2Fsvg%3E";

Ye.div`
  & > * {
    margin: 0 9px;
  }
`;
const ContactSection = Ye.section`
  display: grid;
  row-gap: 18px;
`;
const AlbumSection = Ye.section``;
const AlbumTitle = Ye.div`
  justify-self: stretch;
  display: flex;
  justify-content: space-between;
  align-items: center;

  & > a {
    text-decoration: none;
    font-size: 1.4rem;
    color: #4F9DDE;
    &::after {
      display: inline-block;
      content: url(${arrowRight});
      margin-left: 2px;
    }
  }
`;
const Album = Ye.div`
  margin-top: 14px;
  justify-self: start;
  width: 100%;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 10px;
`;
const Photo = Ye.img`
  width: 76px;
  height: 76px;
  object-fit: cover;
`;
const CloseIcon = Ye(Icon).attrs({
  opacity: 0.3
})`
  position: absolute;
  right: 30px;
  top: 30px;
  cursor: pointer;
`;
const StyledProfile = Ye.div`
  display: grid;
  align-content: start;
  justify-content: center;
  justify-items: center;
  position: relative;
  padding: 30px;
  height: 100vh;
  margin-top: 2vh;
  overflow-y: auto;
`;

var _StyledButton = Ye(Button).withConfig({
  displayName: "Profile___StyledButton",
  componentId: "sc-1jt0ixn-0"
})(["grid-area:1 / 1 / 3 / 2;align-self:end;margin-left:100px;z-index:10;"]);

function Profile({
  showEditBtn,
  showCloseIcon = true,
  onCloseClick,
  onEdit,
  children,
  ...rest
}) {
  return /*#__PURE__*/React__default.createElement(StyledProfile, rest, showCloseIcon && /*#__PURE__*/React__default.createElement(CloseIcon, {
    icon: SvgCross,
    onClick: onCloseClick
  }), showEditBtn && /*#__PURE__*/React__default.createElement(_StyledButton, {
    size: "52px",
    onClick: onEdit
  }, "Edit"), children);
}

Profile.Description = ({
  label,
  children
}) => /*#__PURE__*/React__default.createElement(Paragraph, null, /*#__PURE__*/React__default.createElement(Text, {
  type: "secondary"
}, label, "\uFF1A"), /*#__PURE__*/React__default.createElement(Text, null, children));

var _StyledAvatar = Ye(Avatar).withConfig({
  displayName: "Profile___StyledAvatar",
  componentId: "sc-1jt0ixn-1"
})(["margin:26px 0;grid-area:1 / 1 / 3 / 2;"]);

Profile.Avatar = ({
  avatarUrl,
  size = '160px',
  status,
  statusIconSize = '25px'
}) => /*#__PURE__*/React__default.createElement(_StyledAvatar, {
  src: avatarUrl,
  size: size,
  status: status,
  statusIconSize: statusIconSize
});

Profile.Contact = ({
  email,
  link,
  phone
}) => /*#__PURE__*/React__default.createElement(ContactSection, null, phone && /*#__PURE__*/React__default.createElement(Profile.Description, {
  label: "Phone"
}, phone), email && /*#__PURE__*/React__default.createElement(Profile.Description, {
  label: "Email"
}, email), link && /*#__PURE__*/React__default.createElement(Profile.Description, {
  label: "Website"
}, link));

Profile.Album = ({
  title = 'Album (0)',
  albumList,
  children
}) => /*#__PURE__*/React__default.createElement(AlbumSection, null, /*#__PURE__*/React__default.createElement(AlbumTitle, null, /*#__PURE__*/React__default.createElement(Text, {
  type: "secondary"
}, title)), /*#__PURE__*/React__default.createElement(Album, null, albumList.map(_ => /*#__PURE__*/React__default.createElement(Photo, {
  src: _.photoUrl,
  alt: _.alt
}))), children);

var _StyledParagraph = Ye(Paragraph).withConfig({
  displayName: "Profile___StyledParagraph",
  componentId: "sc-1jt0ixn-2"
})(["margin-bottom:12px;"]);

var _StyledParagraph2 = Ye(Paragraph).withConfig({
  displayName: "Profile___StyledParagraph2",
  componentId: "sc-1jt0ixn-3"
})(["margin-bottom:18px;"]);

var _StyledParagraph3 = Ye(Paragraph).withConfig({
  displayName: "Profile___StyledParagraph3",
  componentId: "sc-1jt0ixn-4"
})(["margin-bottom:26px;"]);

var _StyledDivider = Ye(Divider).withConfig({
  displayName: "Profile___StyledDivider",
  componentId: "sc-1jt0ixn-5"
})(["margin:30px 0;"]);

Profile.Content = ({
  name,
  address,
  detail,
  children
}) => /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement(_StyledParagraph, {
  size: "xlarge"
}, name), /*#__PURE__*/React__default.createElement(_StyledParagraph2, {
  size: "medium",
  type: "secondary"
}, address), /*#__PURE__*/React__default.createElement(_StyledParagraph3, null, detail), children, /*#__PURE__*/React__default.createElement(_StyledDivider, null));

Profile.Description.propTypes = {
  children: propTypes.any,
  label: propTypes.string
};
Profile.Content.propTypes = {
  children: propTypes.any,
  name: propTypes.string,
  address: propTypes.string,
  detail: propTypes.string
};
Profile.Contact.propTypes = {
  link: propTypes.string,
  phone: propTypes.string,
  email: propTypes.string
};
Profile.propTypes = {
  children: propTypes.any,
  showEditBtn: propTypes.bool,
  showCloseIcon: propTypes.bool,
  onEdit: propTypes.func,
  onCloseClick: propTypes.func
};
Profile.Avatar.propTypes = {
  status: propTypes.oneOf(['online', 'offline']),
  avatarUrl: propTypes.string,
  statusIconSize: propTypes.string,
  size: propTypes.string
};
Profile.Album.propTypes = {
  children: propTypes.any,
  title: propTypes.string,
  albumList: propTypes.arrayOf(propTypes.shape({
    key: propTypes.string,
    photoUrl: propTypes.string,
    alt: propTypes.string
  }))
};

export default Profile;
