// react-chat-app/lib/esm/style-fb8054ef.js
import { _ as _e, Y as Ye } from './styled-components.esm-a60371c1.js';

const typeVariants = {
  primary: _e`
    color: ${({
    theme
  }) => theme.grayDark}
    `,
  secondary: _e`
    color: ${({
    theme
  }) => theme.grayDark};
    opacity: 0.3;
  `,
  danger: _e`
    color: ${({
    theme
  }) => theme.red};
  `
};
const sizeVariants = {
  normal: _e`
      ${({
    theme
  }) => theme.normal}
    `,
  medium: _e`
      ${({
    theme
  }) => theme.medium}
    `,
  large: _e`
      ${({
    theme
  }) => theme.large}
    `,
  xlarge: _e`
      ${({
    theme
  }) => theme.xlarge}
    `,
  xxlarge: _e`
      ${({
    theme
  }) => theme.xxlarge}
    `,
  small: _e`
      ${({
    theme
  }) => theme.small}
    `,
  xsmall: _e`
      ${({
    theme
  }) => theme.xsmall}
    `,
  xxsmall: _e`
      ${({
    theme
  }) => theme.xxsmall}
    `
};
const StyledText = Ye.span`
    font-size: ${({
  size
}) => sizeVariants[size] || sizeVariants.normal};
    ${({
  type
}) => typeVariants[type]};
    ${({
  bold
}) => bold && 'font-weight: 500'};
  `;

export { StyledText as S };
